### PR TITLE
Implements previously empty renegotiate & renegotiate_pending functions

### PR DESCRIPTION
**Pull request by kjav**
This fixes bug https://github.com/pyca/pyopenssl/issues/142, and uses bindings committed in https://github.com/pyca/cryptography/pull/2303

### CODE REVIEW

**Comment by kjav** [comment]
How do I get the checks to be run again? The most recent version of the cryptography bindings were not merged until the checks had already failed.

**Comment by kjav** [comment]
Would committing an empty commit work?

**Comment by alex** [comment]
Yup, an empty commit will work, one of the core developers can also reset
travis.

On Fri, Aug 28, 2015 at 12:52 PM, kjav <notifications@github.com> wrote:

> Would committing an empty commit work?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/pyopenssl/pull/305#issuecomment-135830897>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084

**Comment by kjav** [comment]
Don't worry - I managed to find a typo in a comment, so I commited a fix for that.

File: doc/conf.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def find_version(*file_paths):
 # The encoding of source files.
 #source_encoding = 'utf-8-sig'
 
-# The master toctree document.
+# The master doctree document.
```
Comment (alex) [inline doc/conf.py]: I don't think this is actually correct -- I believe this comment indicates that "index" is the root in teh table of contents.

**Comment by alex** [comment]
This will need some tests before it can be landed. Thanks for working on it!

**Comment by kjav** [comment]
Why is codecov 'waiting for Cl to finish' when it has finished already?

**Comment by kjav** [comment]
I think a test has already been written for this function, but commented out. https://github.com/kjav/pyopenssl/blob/master/OpenSSL/test/test_ssl.py#L3208

**Comment by kjav** [comment]
Why is this build failing? It says the function "SSL_renegotiate_pending" is not a member of the _lib module, but it was added in https://github.com/pyca/cryptography/commit/c6a73cb16f00889b53d368359059141d75a26be0

**Comment by kjav** [comment]
Perhaps the cryptography module needs to be rebuild on the build machine (using `pip install -e .` in the root directory)?

**Comment by reaperhulk** [comment]
Because it's not in a released version. The cryptographyMaster builds (which pull from github) are passing.

I've backported your binding into our 1.0.x tree but until we do the 1.0.1 release this won't pass (and then you'll need to update the PR to require cryptography 1.0.1 in PyOpenSSL anyway).

**Comment by kjav** [comment]
Ah, that makes sense. Would I do that using an #ifdef statement/?

**Comment by reaperhulk** [comment]
No, it will be the version restriction in setup.py. Pure Python. But you shouldn't update this PR again until 1.0.1 is out since that's the blocker now :)

**Comment by kjav** [comment]
ok, I'll remember to do that :)

**Comment by kjav** [comment]
Looks like the latest release of cryptography (1.0.1) has my new bindings in it! Is it acceptable to increase the cryptography requirement from 0.7 to 1.0.1?

**Comment by reaperhulk** [comment]
I'm fine with that, but cryptography 1.0+ made some big installation changes (requiring cffi 1.1+) so we should be aware that some users may have some trouble during the transition.

**Comment by hynek** [comment]
I guess that move is inevitable.  We’ll just have to wait until Travis supports an appropriate PyPy to avoid another huge influx of angry bug reports.

**Comment by hynek** [comment]
It appears that we’re already depending on cryptography 1.0 so if you polish it up we can merge this.

**Comment by hynek** [comment]
@kjav are you still interested in working on this?

**Pull request closed without merging**